AWS: Add aws plugin configuration (#31312)
grafana/grafana: docs/sources/administration/configuration.md

Refer to [LDAP authentication]({{< relref "../auth/ldap.md" >}}) for detailed instructions.

## [aws]

> In this section, core and external AWS plugins can be configured.

Reviewer: ```suggestion
You can configure core and external AWS plugins.
```

Reviewer: * Eliminate text that talks about the documentation meta layer (this section, this document, this paragraph, etc.); instead,  talk about the concepts and tasks directly.
* Replace passive voice with active voice and identify the agent (doer of the action) where possible.

### allowed_auth_providers

> Specify what authentication providers that the AWS plugins allow. For a list of allowed providers, refer to the data-source configuration page for a given plugin. If you configure a plugin by provisioning, only providers that are specified in `allowed_auth_providers` are allowed.

Reviewer 2: I think dropping the extra "that" is better English.
```suggestion
Specify what authentication providers the AWS plugins allow. For a list of allowed providers, refer to the data-source configuration page for a given plugin. If you configure a plugin by provisioning, only providers that are specified in `allowed_auth_providers` are allowed.
```

Author: :+1:

> Specify what authentication providers should be allowed for AWS plugins. The providers will be listed in the data source configuration page for the plugin. If the plugin is configured using provisioning, only providers that are specified in `allowed_auth_providers` will be accepted.

Reviewer: ```suggestion
Specify what authentication providers that the AWS plugins allow. For a list of allowed providers, refer to the data-source configuration page for a given plugin. If you configure a plugin by provisioning, only providers that are specified in `allowed_auth_providers` are allowed.
```
* Remove _should_ to reduce ambiguity. Alternatively, use SHOULD in all caps to make it clear that you are referring to reserved words SHOULD, MUST, etc. per https://tools.ietf.org/html/rfc2119. This can, however, can come across as yelling.
* Remove future tense unless there is a (rare) specific reason to use it: _will be_ --> _is_.
* Notice that the passive at end of this suggestion works well because the object is the center of attention in this phrase, "only providers that are specified in `allowed_auth_providers` are allowed."
* Change "accepted" to "allowed" because it seems like those are simply two verbs to refer to the same thing.

> Options: `default` (AWS SDK Default), `keys` (Access && key secret key), `credentials` (Credentials field), `ec2_IAM_role` (EC2 IAM Role)

Reviewer: ```suggestion
Options: `default` (AWS SDK default), `keys` (Access  and key secret key), `credentials` (Credentials field), `ec2_IAM_role` (EC2 IAM role)
```
Why are there two instances of the word "key" in "key secret key"?

Where do these options appear?

Author: oh that's a typo! they appear in the auth drop down of the data source config page. You can check the CloudWatch config page for example.

Reviewer: I assume there is more than one key: access key and secret key.

> ### assume_role_enabled

Reviewer: Why is `_enabled` baked into this?

Author: The h3 headers in this doc maps to a config option in the [default.ini](https://github.com/grafana/grafana/blob/master/conf/defaults.ini) file. The reason it is suffixed with `_enabled` is because that's how all boolean options are named in this file, so I'm just following the used convention.

Set to false to disable AWS authentication using an assumed role with temporary security credentials. Please refer to [AWS official docs](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) for details about assume roles.

> If this option is disabled, the **Assume Role** and the **External Id** field will be removed from the AWS data source configuration page. If the plugin is configured using provisioning, it will be possible using an assumed role as long as `assume_role_enabled` is set to true.

Reviewer: ```suggestion
If this option is disabled, the **Assume Role** and the **External Id** field are removed from the AWS data source configuration page. If the plugin is configured using provisioning, it is possible to use an assumed role as long as `assume_role_enabled` is set to `true`.
```
Vet this for technical accuracy. I was not sure what "it" meant in, ", it will be possible using an assumed role..."

Author: Looks good, thanks!

Used as the default time zone for user preferences. Can be either `browser` for the browser local time zone or a time zone name from the IANA Time Zone database, such as `UTC` or `Europe/Amsterdam`.

## [expressions]

> 

Author: My markdown editor added this newline here. Keep?

> **Note:** This feature is available in Grafana v7.4 and later versions.

### enabled

> 

Author: Same here...
